frontend/src/modules/EditZing/Components/EditZing.tsx

```diff
@@ -40,18 +45,17 @@ export const EditZing = () => {
       })
   }, [courseId])
 
-  const [selectedGroups, setSelectedGroups] = useState<Group[]>([])
+  const [selectedGroupNumbers, setSelectedGroupNumbers] = useState<number[]>([])
 
   const editSelectedGroups = (group: Group, selected: boolean) => {
     if (selected) {
-      setSelectedGroups([...selectedGroups, group])
+      setSelectedGroupNumbers([...selectedGroupNumbers, group.groupNumber])
     } else {
-      setSelectedGroups(
-        selectedGroups.filter((g) => g.groupNumber !== group.groupNumber)
+      setSelectedGroupNumbers(
+        selectedGroupNumbers.filter((g) => g !== group.groupNumber)
       )
     }
   }
```

Comment:
This solution to the bug is a really good idea, tying the selection to the group ids rather than the actual information itself so the information pulled could be up to date when the information requested is actually needed.